### PR TITLE
add unit tests

### PR DESCRIPTION
**Pull request by heim-lukas**

### CODE REVIEW

**Review by wachterjohannes — COMMENTED** [review]
## Code Review - Unit Tests

### ✅ Sehr gut umgesetzt
- Professionelles Test-Setup (xUnit, Moq, FluentAssertions)
- Authorization-Tests validieren Owner-Checks
- Helper-Methoden für Wiederverwendung
- Arrange-Act-Assert Pattern konsequent
- Edge Cases (einzelner Shooter ohne Team)

### ⚠️ Verbesserungspotenzial
- DateTime.Now → Flaky Tests Risiko
- Null-Check Tests fehlen (wichtig für PR #19 Fix!)
- Validierungs-Tests unvollständig
- Team-Score-Logik oberflächlich getestet

Details in Inline-Kommentaren!

File: schiessmeister-csharp-tests/Controller/OrganizationControllerTests.cs [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+        public OrganizationControllerTests()
+        {
+            _orgRepoMock = new Mock<IOrganizationRepository>();
+            _compRepoMock = new Mock<ICompetitionRepository>();
+
+            _controller = new OrganizationController(_orgRepoMock.Object, _compRepoMock.Object);
+        }
+
+        private void SetUserId(int userId)
```
Comment (wachterjohannes) [inline schiessmeister-csharp-tests/Controller/OrganizationControllerTests.cs:26]: **[Sehr Gut] - Helper-Methode für Authorization-Testing**

Exzellent! `SetUserId()` macht Authorization-Testing sehr einfach und wiederverwendbar.

File: schiessmeister-csharp-tests/Controller/OrganizationControllerTests.cs [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+            result.Result.Should().BeOfType<NotFoundResult>();
+        }
+
+        [Fact]
+        public async Task CreateCompetition_ReturnsCreated_WhenValid()
+        {
+            var ownerId = 42;
+            var newCompetition = new Competition
+            {
+                Id = 0,
```
Comment (wachterjohannes) [inline schiessmeister-csharp-tests/Controller/OrganizationControllerTests.cs:92]: **[Niedrig] - DateTime.Now in Tests**

**Problem**: Flaky Tests möglich bei Verwendung von `DateTime.Now`.

**Empfehlung**:
```csharp
var now = new DateTime(2025, 1, 1, 10, 0, 0); // Festes Datum
StartDateTime = now,
EndDateTime = now.AddHours(2)
```

File: schiessmeister-csharp-tests/Controller/OrganizationControllerTests.cs [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+            };
+
+            // Act
+            var result = await _controller.CreateCompetition(1, comp);
+
+            // Assert
+            result.Result.Should().BeOfType<BadRequestObjectResult>();
+            ((BadRequestObjectResult)result.Result!).Value.Should().Be("At least one discipline is required.");
+        }
+
```
Comment (wachterjohannes) [inline schiessmeister-csharp-tests/Controller/OrganizationControllerTests.cs:139]: **[Exzellent] - Authorization Tests**

Sehr wichtig! Diese Tests validieren die Owner-Checks. Zeigt dass das Team Security ernst nimmt.

File: schiessmeister-csharp-tests/Service/LeaderboardServiceTests.cs [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+        private readonly Mock<ICompetitionRepository> _repoMock;
+        private readonly LeaderboardService _service;
+
+        public LeaderboardServiceTests()
+        {
+            _repoMock = new Mock<ICompetitionRepository>();
+            _service = new LeaderboardService(_repoMock.Object);
+        }
+
+        // Helper to create ShootingResult
```
Comment (wachterjohannes) [inline schiessmeister-csharp-tests/Service/LeaderboardServiceTests.cs:20]: **[Sehr Gut] - Helper für Test-Daten**

Gute Abstraktion mit `CreateResult()`! Reduziert Duplikation in Tests.

File: schiessmeister-csharp-tests/Service/LeaderboardServiceTests.cs [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+                {
+                    new ShootingSeries
+                    {
+                        Points = new[] { totalPoints }
+                    }
+                }
+            };
+        }
+
+        [Fact]
```
Comment (wachterjohannes) [inline schiessmeister-csharp-tests/Service/LeaderboardServiceTests.cs:35]: **[Mittel] - Fehlende Null-Tests**

**Erinnerung aus PR #17**: `participation.Result.TotalPoints` kann crashen wenn Result null ist!

**Empfehlung**: Test hinzufügen der den Fix aus PR #19 validiert:
```csharp
[Fact]
public async Task GetLeaderboards_ThrowsException_WhenShooterIsNull()
{
    var competition = new Competition
    {
        Participations = new[] {
            new Participation { Shooter = null } // ← NULL!
        }.ToList()
    };

    Func<Task> act = async () => await _service.GetLeaderboardsAsync(competition);

    await act.Should().ThrowAsync<InvalidOperationException>()
        .WithMessage("*Some participations have no shooter*");
}
```

File: schiessmeister-csharp-tests/schiessmeister-csharp-tests.csproj [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    <RootNamespace>schiessmeister_csharp_tests</RootNamespace>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+
+    <IsPackable>false</IsPackable>
+    <IsTestProject>true</IsTestProject>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="coverlet.collector" Version="6.0.0" />
```
Comment (wachterjohannes) [inline schiessmeister-csharp-tests/schiessmeister-csharp-tests.csproj:14]: **[Exzellent] - Moderne Testing-Pakete**

Perfekte Auswahl:
- xUnit: Industry Standard für .NET
- Moq: Repository-Mocking
- FluentAssertions: Lesbare Assertions
- coverlet: Code Coverage
